### PR TITLE
Synchronize data context changes

### PR DESCRIPTION
**Pull request by sfentress**
This synchronizes multiple user's data contexts according to the following rules:

1. If a user joins a share with a new table, the new table will exactly match the structure of the shared table when it is first created
2. If a user joins a share with an existing table, the user's own structure is preserved, but any new attributes are added to the shared data context in both directions, so that all parties will end up with the same attributes. No attributes are deleted.
3. From then on, users' own structures are preserved, but attribute adding and deletion is synchronized between users
4. When attributes are added, they are added to a collection of the same name if one exists, otherwise they are added at the end of the last collection

### CODE REVIEW

**Comment by sfentress** [comment]
> if left shares a table with attribute a, and right joins with an existing table with attribute b, attribute b is deleted rather than being shared, which is what I expected.

That's concerning. However, I don't see it. Can you double-check?

My steps:

1. Left creates a table in CODAP, names the first attribute "a", and shares that table
2. Right creates a table in CODAP, names the first attribute "b", and joins the share
3. My result: both users see "a" and "b"

The other suggestions look good.

File: src/lib/db.ts [hunk 1]
```diff
@@ -99,6 +101,10 @@ export class DB {
     if (this.unregisterPresence) {
       this.unregisterPresence();
     }
+    if (this.shareRef) {
```
[the remark this answers is not in the document]
Comment (sfentress) [reply src/lib/db.ts]: Yes, poor branch rebasing.

**Comment by kswenson** [comment]
My scenario was different, but I'm seeing it with yours as well. I'll see if I can figure out the difference.

**Comment by sfentress** [comment]
@kswenson I believe my latest commit fixes the issue.

**Pull request merged**